### PR TITLE
Clean up View and related ops

### PR DESCRIPTION
**Pull request by desertfire**
Summary:
- Rename View to LazyView to avoid name conflict with aten::view
- Move view ops to a separate directory
- Define internal view opcode
- Unify the naming and file location for "reverse" view ops

### CODE REVIEW

File: lazy_tensor_core/lazy_tensor_core/csrc/lazy_view.cpp [hunk 1]
```diff
@@ -74,14 +71,14 @@ torch::lazy::Value ApplyUpdate(torch::lazy::Value ir_value,
     const ViewInfo& view_info = update_data.view_infos[i - 1];
     switch (view_info.view_type) {
       case ViewInfo::Type::kSelect:
-        result = torch::lazy::MakeNode<ir::ops::Unselect>(
+        result = torch::lazy::MakeNode<ir::ops::SelectReverse>(
```
Comment (wconstab) [inline lazy_tensor_core/lazy_tensor_core/csrc/lazy_view.cpp:74]: this and GenericSliceReverse - just renaming the same things, but for more clarity?

File: lazy_tensor_core/lazy_tensor_core/csrc/ts_backend/ts_node_lowering.cpp [hunk 2]
```diff
@@ -77,36 +72,35 @@ class TSNodeLowering : public TSNodeLoweringInterface {
...
+      return LowerAsStridedReverse(
+          torch::lazy::NodeCast<torch_lazy_tensors::ir::ops::AsStridedReverse>(
+              node, torch_lazy_tensors::ir::ops::as_strided_reverse));
     }
     if (node->op() == *torch_lazy_tensors::ir::ops::ltc_cast) {
       return LowerCast(torch::lazy::NodeCast<torch_lazy_tensors::ir::ops::Cast>(
           node, *torch_lazy_tensors::ir::ops::ltc_cast));
     }
-    if (node->op() == *torch_lazy_tensors::ir::ops::ltc_generic_slice) {
+    if (node->op() == torch_lazy_tensors::ir::ops::generic_slice) {
```
Comment (wconstab) [inline lazy_tensor_core/lazy_tensor_core/csrc/ts_backend/ts_node_lowering.cpp:84]: i'm wondering.. (and it's fine for this to be a separate PR if we do it) - should we move these lowerings into the node classes?  hm, i guess maybe we can't; i have not read the node classes yet, but since they are not under ts_backend/ it probably doesn't make sense to include lowerings in them.

File: lazy_tensor_core/lazy_tensor_core/csrc/ts_backend/ts_shape_inference.cpp [hunk 3]
```diff
@@ -104,15 +102,15 @@ torch::lazy::Shape InferStack(const ir::ops::Stack* stack) {
...
     auto generic_slice = torch::lazy::NodeCast<ir::ops::GenericSlice>(
-        node, *ir::ops::ltc_generic_slice);
+        node, ir::ops::generic_slice);
     const torch::lazy::Output& argument = node->operand(0);
     return torch::lazy::Shape(
         torch::lazy::GetShapeFromTsOutput(argument).scalar_type(),
         generic_slice->sizes());
   }
-  if (node->op() == *ir::ops::ltc_update_slice) {
+  if (node->op() == ir::ops::generic_slice_reverse) {
```
Comment (wconstab) [inline lazy_tensor_core/lazy_tensor_core/csrc/ts_backend/ts_shape_inference.cpp:113]: what's up with dropping the * operator here?

File: lazy_tensor_core/lazy_tensor_core/csrc/view_ops/as_strided.h [hunk 4]
```diff
@@ -1,14 +1,14 @@
...
 #include <vector>
 
+#include "lazy_tensor_core/csrc/view_ops/opcode.h"
+
 namespace torch_lazy_tensors {
 namespace ir {
 namespace ops {
 
-class AsStrided : public torch::lazy::TsNode {
+class AsStrided : public BaseNode {
```
Comment (wconstab) [inline lazy_tensor_core/lazy_tensor_core/csrc/view_ops/as_strided.h:11]: wdyt about adding a README to the view_ops folder?  I think at least 2 things would be good to explain in general:
 - how this 'BaseNode' works : haven't found it yet but I assume you typedef it somewhere and it is TsNode in our case...
 - what the pattern of {View} and {View}Reverse ops is about

Comment (wconstab) [reply lazy_tensor_core/lazy_tensor_core/csrc/view_ops/as_strided.h:11]: (could also be in a central view.h header or something, instead of readme...)

File: lazy_tensor_core/lazy_tensor_core/csrc/view_ops/opcode.h [hunk 5]
```diff
@@ -0,0 +1,28 @@
...
+
+#include <torch/csrc/lazy/ts_backend/ts_node.h>
+
+namespace torch_lazy_tensors {
+namespace ir {
+namespace ops {
+
+// TODO: Allow other backend to define its own BaseNode, or
+// promote TsNode to Node.
+using BaseNode = torch::lazy::TsNode;
```
Comment (wconstab) [inline lazy_tensor_core/lazy_tensor_core/csrc/view_ops/opcode.h:11]: ah.. here it is.  Maybe if you rename this file to ... ViewBase?  (and maybe even rename BaseNode to ViewBase?) it would become a little more discoverable without explanation

File: lazy_tensor_core/lazy_tensor_core/csrc/view_ops/opcode.h [hunk 6]
```diff
@@ -0,0 +1,28 @@
+#pragma once
+
+#include <torch/csrc/lazy/ts_backend/ts_node.h>
+
+namespace torch_lazy_tensors {
+namespace ir {
+namespace ops {
+
+// TODO: Allow other backend to define its own BaseNode, or
```
Comment (wconstab) [inline lazy_tensor_core/lazy_tensor_core/csrc/view_ops/opcode.h:9]: so, the idea would be that we replace this part with a generated header, so that the IR codegen (which already knows about the custom backend node class) just stamps out the right `using` statement, right?

I was thinking it's nicer to leave this file as a non-generated one, and have it include the generated file- this is more discoverable to users...

But i'm afraid that's impossible unless we actually codegen this file too, since the include path would be different for xla backend and ts backend.  Unless there is some way to make the xla codegen (which runs out of tree) generate a header in the torch/csrc/lazy/core/ tree directly instead of in the torch_xla external tree?

I'm actually not sure how this would work for xla.  Guess we better figure it out before we go this route, generally

**Review by wconstab — COMMENTED** [review]
Overall looks great! But I ran into one issue about how we'd codegen the custom view base - see my question inline.  If you have a plan for that then i'm happy.

File: lazy_tensor_core/lazy_tensor_core/csrc/ts_backend/ts_shape_inference.cpp (hunk 3 above)
Comment (desertfire) [reply lazy_tensor_core/lazy_tensor_core/csrc/ts_backend/ts_shape_inference.cpp:113]: I changed the declaration of this op into plain OpKind (see opcode.h) instead of OpKindWrapper. I don't see a value   of using OpKindWrapper.

File: lazy_tensor_core/lazy_tensor_core/csrc/lazy_view.cpp (hunk 1 above)
Comment (desertfire) [reply lazy_tensor_core/lazy_tensor_core/csrc/lazy_view.cpp:74]: To make them more consistent across view ops. Before there are `AsStridedViewUpdate`, `DiagonalViewUpdate`, `UpdateSlice` and `Unselect`, and I have changed them into `*Reverse`.

File: lazy_tensor_core/lazy_tensor_core/csrc/ts_backend/ts_node_lowering.cpp (hunk 2 above)
Comment (desertfire) [reply lazy_tensor_core/lazy_tensor_core/csrc/ts_backend/ts_node_lowering.cpp:84]: Right, if later we decide to promote TsNode to Node, then it makes sense to make these as node's Lower methods. For now, we probably should just keep them here.

File: lazy_tensor_core/lazy_tensor_core/csrc/view_ops/as_strided.h (hunk 4 above)
Comment (desertfire) [reply lazy_tensor_core/lazy_tensor_core/csrc/view_ops/as_strided.h:11]: Ok, let me add some description in the new `opcode.h`

**Comment by desertfire** [comment]
Close this PR to favor a breakdown of smaller PRs. Also per our discussion, PyTorch/XLA will not be blocked on this version of LazyView merge, and they will prefer to work on a later solution that is built on top of PyTorch core's functionalization pass.
